graylog2-web-interface/src/components/common/DataTable/DataTable.jsx

```diff
@@ -197,17 +197,18 @@ class DataTable extends React.Component {
 
     return (
       <div>
-        <Filter customFilter={customFilter}
-                label={filterLabel}
-                id={id}
-                rows={rows}
-                displayKey={displayKey}
-                filterBy={filterBy}
-                filterSuggestions={filterSuggestions}
-                filterKeys={filterKeys}
-                onDataFiltered={this.filterDataRows}>
-          {children}
-        </Filter>
+        {customFilter || (
+          <Filter label={filterLabel}
+                  rows={rows}
+                  id={id}
+                  displayKey={displayKey}
+                  filterBy={filterBy}
+                  filterSuggestions={filterSuggestions}
+                  filterKeys={filterKeys}
+                  onDataFiltered={this.filterDataRows}>
+            {children}
+          </Filter>
+        )}
```

Comment:
I tried to implement the following approach:
```
const TableFilter = customFilter ?? Filter

return (<div> 
  <TableFilter
      ...
```

This way the `customFilter` has access to all props the `Filter` component receives. This caused a problem where the filter input value got removed after submitting the filter. I tried to fix the problem with `useCallback` / `useMemo`, but due to the dependency chain, I would have to split up the pagination state of every component which uses the `DataTable` with a `customFilter`.